### PR TITLE
chore(deps): upgrade dependencies

### PR DESCRIPTION
**Pull request by sam-goodwin**
Upgrades project dependencies. See details in [workflow run].

[Workflow Run]: https://github.com/sam-goodwin/typesafe-dynamodb/actions/runs/6568098742

------

*Automatically created by projen via the "upgrade-main" workflow*

### CODE REVIEW

**Comment by 1oglop1** [comment]
@sam-goodwin Any chance you can revive this? The type error is back.
And I am getting warnings
`warning workspace-aggregator-30b17005-996c-45b2-8f02-6706a6a05406 > @myPackage.lib > typesafe-dynamodb > @aws-sdk/smithy-client@3.374.0: This package has moved to @smithy/smithy-client`

File: package.json [hunk 1]
```diff
@@ -23,31 +23,31 @@
...
-    "@aws-sdk/client-dynamodb": "^3.100.0",
-    "@aws-sdk/lib-dynamodb": "^3.100.0",
-    "@aws-sdk/smithy-client": "^3.99.0",
-    "@aws-sdk/types": "^3.78.0",
-    "@aws-sdk/util-dynamodb": "^3.100.0",
-    "@types/aws-lambda": "^8.10.97",
-    "aws-sdk": "^2.1145.0"
+    "@aws-sdk/client-dynamodb": "^3.378.0",
+    "@aws-sdk/lib-dynamodb": "^3.378.0",
+    "@aws-sdk/smithy-client": "^3.370.0",
```
Comment (1oglop1) [inline package.json:46]: According to the message, I think this will need a replacement for @smithy/smithy-client

Comment (sam-goodwin) [reply package.json:46]: I took a look at fixing this PR. Ran into some critical type error when upgrading to Ts 5 and latest smithy. Needs some dedicated attention. I'll try and find some time this weekend to fix!

Comment (1oglop1) [reply package.json:46]: no need to stress it, I'm glad that it will eventually work.

**Pull request closed without merging**